### DISCUSSION TITLE
db encryption gone wrong

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2503] [Q&A] [answered]
I am trying to follow this for **windows**:
https://drift.simonbinder.eu/docs/other-engines/encryption/#encrypted-version-of-a-nativedatabase
and I had `sqlite3_flutter_libs` in my yaml so I replace it with `sqlcipher_flutter_libs` as instructed and now I get the following error:

```
CMake Error at C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.26/Modules/FindOpenSSL.cmake:670 (find_package_handle_standard_args)
  flutter/ephemeral/.plugin_symlinks/sqlcipher_flutter_libs/windows/CMakeLists.txt:27 (find_package)
```

I tried flutter clean and updating visual studio and other random stuff but nothing works

can anybody help? not sure what the problem is

**Comment by simolus3** [comment, marked as answer]
As the [readme](https://github.com/simolus3/sqlite3.dart/tree/main/sqlcipher_flutter_libs) points out (although not very prominently, my bad), you need to have static OpenSSL libraries to link available on your development PC. We need those to compile SQLCipher.

You can obtain them via a package manager (`choco install openssl` with Chocolatey) or [via other means](https://wiki.openssl.org/index.php/Binaries) or see [this](https://teskalabs.com/blog/openssl-binary-distribution-for-developers-static-library).

**Reply by BananaMasterz** [reply]
ah you're write... I didn't read the readme since I was following the instructions above a bit too blindly lol thanks tho!
